fix to latest solc

3esmit opened the pull request.

3esmit committed:
fix to latest solc

chriseth commented on data.sol:
```diff
@@ -1,4 +1,4 @@
-pragma solidity ^0.5.0;
```
Is this a semantic change?

3esmit replied:
I started using this new pragma because I saw it on the docs. It does not make differece right now, but seems more complete.

3esmit replied:
This is how solidity style guide suggests https://solidity.readthedocs.io/en/v0.5.2/style-guide.html

chriseth commented on patricia.sol:
```diff
@@ -13,40 +13,42 @@ contract PatriciaTree {
-            var (prefix, suffix) = Utils.splitCommonPrefix(k, e.label);
+            D.Label memory prefix;
+            D.Label memory suffix;
+            (prefix, suffix) = Utils.splitCommonPrefix(k, e.label);
```
You should be able to use
```suggestion
            (D.Label memory prefix, D.Label memory suffix) = Utils.splitCommonPrefix(k, e.label);
```

chriseth commented on patricia.sol:
```diff
@@ -55,7 +57,9 @@ contract PatriciaTree {
-            var (head, tail) = Utils.chopFirstBit(suffix);
+            uint256 head;
+            D.Label memory tail;
+            (head, tail) = Utils.chopFirstBit(suffix);
```
You should be able to use
```suggestion
            (uint head, D.Label memory tail) = Utils.chopFirstBit(suffix);
```

chriseth reviewed (changes requested):
Thanks a lot!

I think this is generally fine, but I would recommend some style changes:
 - keep pragma as it was (unless I'm mistaken about something)
 - declare variables with tuple assignment (more places than what I highlighted)

chriseth committed:
Update patricia.sol

Co-Authored-By: 3esmit <3esmit@gmail.com>

chriseth committed:
Update patricia.sol

Co-Authored-By: 3esmit <3esmit@gmail.com>

3esmit committed:
tuple assignment, require messages, hash func to pure

3esmit committed:
tuple assignment, require messages,

3esmit committed:
reoder functions as style guide

3esmit commented:
I did the changes you requested, and also some things I missed (like hash function as view instead of pure). 
The pragma I changed for how it is defined on Solidity Style Guide. 
Also fixed the ordering of functions (external - public - internal - private).

chriseth commented:
Thanks a lot!
